src/scripts/shouldCallAPI.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+function shouldCallAPI(state, expiresInSeconds) {
```

Comment:
This is a bit more of a "usage" of the middleware than a feature of the middleware itself. It assumes a bit too much about the dependent for a packaged library to have ownership over. I could see an "examples" section showing how `shouldCallAPI` can work with illustrations like this. However it currently crosses into territory that the library is otherwise not responsible/aware of